dev/buildtool/bom_commands.py

```diff
@@ -182,11 +183,29 @@ def build(self):
 
     services = dict(self.__base_bom.get('services', {}))
     changed = False
+    def to_semver(build_version):
+      index = build_version.find('-')
+      return build_version[:index] if index >= 0 else build_version
+
     for name, info in self.__services.items():
+      labels = {'repostiory': name, 'branch': options.git_branch, 'updated': True}
       if info['commit'] == services.get(name, {}).get('commit', None):
-        logging.debug('%s commit hasnt changed -- keeping existing %s',
-                      name, info)
-        continue
+        if to_semver(info['version']) == to_semver(services.get(name, {}).get('version', '')):
```

Comment:
This check here is the actual fix.
The other changes are mostly preserving monitoring, though I'm not sure if the metrics are still of use.